### PR TITLE
Show object creation intellisense when new is typed on the line befor…

### LINKED ISSUE #15804: No intellisense after new in object creation expression [attributes case]
**Version Used**: 2.0.0.61205

**Steps to Reproduce**:

```C#
class Program
{
    Program p = new |

    [STAThread]
    static void Main() { }
}
```

**Expected Behavior**: `Program` preselected in autocomplete list

**Actual Behavior**: No intellisense at all.

cf. #2836 #12755 #13402 #14084 
/cc @rchande

### PR DESCRIPTION
**Pull request by rchande**
…e an attribute.

Fixes #15804

Tag @dotnet/roslyn-ide for review

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -1198,6 +1199,11 @@ private IEnumerable<TypeInferenceInfo> InferTypeInIfStatement(IfStatementSyntax
                 return SpecializedCollections.SingletonEnumerable(new TypeInferenceInfo(this.Compilation.GetSpecialType(SpecialType.System_Boolean)));
             }
 
+            private IEnumerable<TypeInferenceInfo> InferTypeInImplicitArrayCreation(ImplicitArrayCreationExpressionSyntax implicitArray, SyntaxToken token)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: "previousToken" instead of "token.

**Comment by CyrusNajmabadi** [comment]
👍 (just change the parameter name).

**Comment by rchande** [comment]
**Ask Mode Info**
- **Customer scenario**: Customer is trying to type an object creation expression one a line before an attribute usage. The `new` keyword is parsed as part of an implicit array creation with the attribute on the following line, which means object creation preselection does not work.
Bugs this fixes**: #15804 
- **Workarounds**: None
- **Risk**: Low
- **Performance impact**: None--no allocation/complexity changes.
- **Is this a regression from a previous update?**: No
- **How was the bug found?**: Customer feedback

**Comment by rchande** [comment]
retest windows_eta_open_prtest please

**Comment by rchande** [comment]
Tag @srivatsn for ask mode

**Pull request merged**